### ISSUE TITLE
Internationalization fails with some locales, for components wrapped by IntlProvider of their own

### ISSUE THREAD

**Issue by planetcrypton** [commercetools/ui-kit#2214]
**Describe the bug**
The bug appeared after upgrading our packages:
When components that use internationalization (useIntl/IntlProvider), such as `DateInput`, are ran isolated, they adhere to the nearest `IntlProvider` perfectly according to its `locale`-value; shows date in expected format, weekdays in correct language etc.
But when ran within the CT `ApplicationShell`, and you provide an `IntlProvider` of your own (meaning it is nested into the CT `ApplicationShell`'s `IntlProvider` somewhere) it's not always like that (only for some specific locale-values such as 'de', 'fr', 'es', but not for eg 'da', 'nl', 'en-gb', 'en-au'). In those cases they fall back to default locale 'en'.

**To Reproduce**

Packages:
"@commercetools-frontend/application-shell": "21.5.1"
"@commercetools-uikit/date-input": "^15.0.2"

1. Create a component using `<DateInput value="2022-06-30" onChange={() => {}} />`
2. Wrap it with one (or multiple, if you like) IntlProviders: eg `<IntlProvider locale="da">`
3. Run it isolated _not_ being wrapped by `ApplicationShell`. (eg in StoryBook)
You'll see the date displayed like this, when the (nearest) IntlProvider has:
locale="da" : 30.06.2022
locale="de" : 30.06.2022
locale="de-at" : 30.06.2022
locale="nl" : 30-06-2022
locale="fr" : 30/06/2022
locale="es" : 30/06/2022
locale="en" : 06/30/2022
locale="en-us" : 06/30/2022
locale="en-gb" : 30/06/2022
locale="en-au" : 30/06/2022
4. Now run this component in CommerceTools' `ApplicationShell`, and you will see:
locale="da" : 06/30/2022 // falls back to default locale 'en'.
locale="de" : 30.06.2022
locale="de-at" : 30.06.2022
locale="nl" : 06/30/2022 // falls back to default locale 'en'.
locale="fr" : 30/06/2022
locale="es" : 30/06/2022
locale="en" : 06/30/2022
locale="en-us" : 06/30/2022
locale="en-gb" : 06/30/2022 // falls back to default locale 'en'.
locale="en-au" : 06/30/2022 // falls back to default locale 'en'.

**Expected behavior**
What I expected was the component ran in CommerceTools' `ApplicationShell` (step 4) to show dates formatted by the given locale and translate weekdays etc, as it does in isolated mode (step 3), just as it did before we upgraded. The previous versions were:
"@commercetools-frontend/application-shell": "17.3.1"
"@commercetools-uikit/date-input": "10.38.0"

**Additional context**
Disclaimer for this nested IntlProvider hack:
Since our organization is located in Denmark (and danish is not an option in CommerceTools' user-profile settings, do you happen to have any plans for that ..🙏..?? Salling Group would be delighted 😀) we'd like people to read and write dates (in a non-US-english date-format) with day first, month second.
Setting the user-profile language to 'en-GB', unexpectedly still comes with the US formatted date; hence this hack...

**Comment by CarlosCortizasCT** [comment]
Hello @planetcrypton,

We've been looking at this problem but it seems we would need some more context.

First of all, when we select `en-GB` as a language in the Merchant Center, we can see the right date formatting for that locale (DD/MM/YYYY) so we're not sure if we are testing the same way you do.

Are you selecting the language in the user profile?
![image](https://user-images.githubusercontent.com/97907068/174745243-4f685dbc-a14f-4a48-bc5c-a48ebcb512bb.png)

On the other hand, the `IntlProvider` does not directly control how dates are formatted. We use [`moment`](https://momentjs.com/) library for dates handling and the provider only _tells_ `moment` which locale it should use when formatting.
Moreover, `moment` have specific packages for data localization and we do not load all of them as the list is huge. So, when the `ApplicationShell` is first rendered, is loads the data localization for the selected language (our supported ones are: en, en-GB, de, fr, es, ja and zh-cn).

Perhaps you could provide any other information about your context so we can better understand your problem.

Thanks.

**Comment by planetcrypton** [comment]
Hi @CarlosCortizasCT ,

Regarding the "en-GB" user-profile language:
I managed to find out what was wrong at our side 😃, after you pointed me in the right direction 🙏!
Yes, I was referring to the same user-profile language as you.
I created a fresh new app using `create-mc-app` and "en-GB" displayed dates correctly ("dd/mm/yyyy"), in contrast to our production code-base (mm/dd/yyyy), even though all `@commercetools*` packages had the same version. So the bug must be related to difference in setup:
Our production code-base uses (within Rush) PNPM with `--strict-peer-dependencies` that will fail if there are unsatisfied peer dependencies. `moment`, as you mentioned, is defined as a peer dependency of among others `@commercetools-uikit/date-input`. Therefor we are forced to install that package ourselves. Then I noticed that our moment version was behind compared to the `create-mc-app` app's code-base (`yarn why moment`). After updating `moment` from `2.29.1` to `2.29.3` that issue was solved and we are able to use "en-GB" as the user-profile language, formatting the dates correctly. Our actual problem is now solved, thanks!

We can now remove our nested `IntlProvider`s (hack).

Back to the title of this issue: Updating `moment` does not solve `DateInput`s within a nested `<IntlProvider locale="da">` to show danish dates and calendar, as we could previously - but that probably has to do with what you explained:
> when the ApplicationShell is first rendered, is loads the data localization for the selected language (our supported ones are: en, en-GB, de, fr, es, ja and zh-cn).

And we don't need such hack anymore 😀
